pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -994,3 +994,30 @@ def test_groupby_get_by_index():
     res = df.groupby("A").agg({"B": lambda x: x.get(x.index[-1])})
     expected = pd.DataFrame(dict(A=["S", "W"], B=[1.0, 2.0])).set_index("A")
     pd.testing.assert_frame_equal(res, expected)
+
+
+def test_aggregate_categorical_with_isnan():
+    # GH 29837
```

Comment:
prob should have been in test_category.py

Reply:
There are two other tests for categorical in that file, test_agg_cython_category_not_implemented_fallback and test_aggregate_categorical_lost_index. move those as well?

Reply:
yeah prob

Reply:
moved in #35052